### DISCUSSION TITLE
Koriki for RG35XX Beta 0.8

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#10] [Announcements]
# Introduction
This is the second public release of Koriki for the RG35XX. Note that this is a beta release, so while the system is usable there may be issues that still need to be solved. Use the [Discussions Page](https://github.com/rg35xx-cfw/Koriki/discussions) to discuss any issues you may find, and create issues in the [Issues Page](https://github.com/rg35xx-cfw/Koriki/issues)

# Installation:
* Flash the koriki-rg35xx image to a SDCARD using balena etcher or a similar software
* Boot with the SDCARD. The first time the firmware will expand the second partition and copy some firmware files to that partition
* Important: this version now uses FAT32 for the second partition, so it should be accessible on Mac/PC without any dedicated software
* If you use a second SDCARD, the system will install those files in the last partition of the second card
* Note that the first time the system boots it will take a bit longer to expand the card and copy files over. Even when the menu is up, there will be a few seconds when the system will be a bit slow because it's adding the virtual memory image
* After the first boot the system will create some configuration folders as well as the roms and bios folders. Add your roms to the roms folder, and your bios to the roms folder.

# Update from previous installations:

* NOTE: If you installed v0.7 or higher, it's safe to use the update file. If you installed v0.6 or a previous version, you should install the full image instead since the partition tables have changed.
* Uncompress the file boot.tar.xz, and replace the contents of the first partition of your SDCARD image with this one (you should see files like uImage, ramdisk.img, boot folder, etc.)
* If you run into issues with content not starting or any other issue, rename your system folder (from the last partition of your main SDCARD or your second SDCARD), and start again.

# IMPORTANT
* If you use a second SDCARD that you have previously used with koriki for RG35XX, please rename your system folder, and start again.
* Some RG35XX boards have issues booting koriki. Download the archive ``bootfix_for_some_revisions.zip``, extract it, and copy the files ``uImage`` and ``kernel.dtb`` to the first Koriki SDCARD partition (overwriting the existing ones).

# Charger behavior:
* The charging logic is slightly different from the one in the stock firmware (or Garlic). The system reimplements the charger logic that is performed via software and not via hardware in this system.
* Shutdown:
  * If the charger power cable is connected to the USB C port, once the system is shutdown it will boot into the charger mode. You will briefly see a battery symbol and the system will seem to be off but the charger led will be on (orange). To boot from this state, briefly press power key.
  * If the charger cable is not connected, the system will be off completely. To boot again from this state you need to press the power button a few seconds and the system will boot.
* Reboot:
  * The system will reboot as expected
 
# ChangeLog:
2023/06/19
* Bumped version to Beta 0.8
* Added timeout for ``POWER`` button press. System no longer shutdowns immediately when the button is pressed, now it needs to be pressed about 3s
* Fixed volume increasing to 100% when unit was shut down via power button
* Fixed brightness control combo (``MENU``+``VOL+/-``)
* Fixed audio volume keys if you use adb_uImage (mixer was different between adb_uImage and uImage)
* Fixed HDMI audio. Note that HDMI cable needs to be plugged in before boot for HDMI to work
* Added PCSX standalone configuration
* Added Devolutionx (Diablo) gamepad support
* Re-Added KM MAME 2003 xtreme core (removed by mistake in 0.7.3)
* Added libretro CROCODS core (Amstrad CPC)
* Added libretro quickness core (NES)
* Added libretro bk (elektronica PDP)
* Added libretro gearcoleco (ColecoVision)
* Added libretro GME (blargg's Game_Music_Emu)
* Added libretro RACE (NGPC)
* Added launcher for eduke32 (DukeNukem3D)

2023/06/16
* Modified the FP flags from VFP3-D16 to NEON due to some performance issues detected in PCSX rearmed core. This should bring some performance improvements
* Recompiled all cores with the new architecture
* Updated full buildroot packages to synchronize to the latest batocera build
* Removed some unnecessary patches. Note that there may be some inconsistencies with previous releases, please report those in the discussions page, discord, or create issues if the problem is well defined
* Improved charging logic. When connecting a power cable when the console is off, the console will enter into charging mode without booting into koriki
* Added PCSX standalone (note, currently is not possible to exit the emulator, if anyone knows how to quit, please let me know)
* Added default mupen64 controls (you can exit the emulator with ``START``+``SELECT``)
* Added ATC theme in Spanish (Thanks to @albgr1979) (theme switching should now work)
* Enabled USB for controllers and some other devices. By default this release works with USB devices like gamepads, USB Wifi and USB Ethernet. However, please not that USB wireless is not currently configured.
  * Note: If you want to use ADB, copy over ``adb_uImage`` into  ``uImage`` 
* Improved input handling by updating the input driver to the latest garlic gpio module (quick movement from L to R or U to D should not leave the key stuck in the initial direction)
  * Note: there's a regression with the power button due to the new input driver. Use the simplemenu menu to shutdown/reboot

2023/06/13
* Added power button shutdown. Short press power button will shutdown the console. If the charger is connected it will boot into charger mode. From there another short press boots the console
* Fixed shutdown while charging
* Added brightness control shortcut with ``MENU``+``VOLUME+/VOLUME-``
* Fixed Sega32X launcher
* Changed AmstradCPC core to cap32

2023/06/12
* Update to fix an issue with the swap and share partitions being overwritten
* Reapplied the SDL patch that was not correctly applied (thanks again @johnnyonflame)
* Fixed typo in mastersystem path
* Added proper openbor launchers
* Added o2em launcher
* Fixed sega32x path

2023/06/11
* Added SNES mednafen-supafaust core (default for SNES). StarFox now runs close to 60fps without overclock or video threading
* Added MAME km_mame_2003_xtreme that runs much better for some roms (default for MAME)
* Fixed swap virtual memory generation, it has its own dedicated partition now, should be much more efficient and not impact FAT32 partitions
* Added additional ports location
* Added SDL VT patch to solve some issues with some ports not returning to simplemenu (thanks @johnnyonflame)

2023/06/08
* Fixed simplemenu NO GAMES FOUND message. Now it will go back to the last selection after a game is closed or after restart
* Changed data partition to FAT32 format
* Added controller configuration for standalone emulators (flycast, PPSSPP, mupen64plus, and Drastic)
* Changed default emulators, removed some missing emulators from the core list

2023/06/06
* First public release (alpha v0.5)
* Implemented global audio control (thanks to [adixal](https://codeberg.org/adixal) for his muOS joyaudio driver)
* Improved the charger logic. System now should properly respond to shutdown or reboot commands in koriki/simplemenu. Just go to settings -> Session, press left/right until you see either shutdown or reboot and press A to confirm. See charger behavior below for details.
* Updated flycast and ppsspp cores to the latest release
* Added drastic, flycast, and ppsspp application launchers. 

2023/06/04
* Updated mgba core
* Added cores: vba-m, gpsp, stella2014
* Added flycast standalone
* Fixed partition autoexpand

2023/06/03
* Updated launchers and menu sections so most systems should now work right out of the box
* Fixed charger led indicators
* Added od-commander
* Added missing flycast_libretro cores
* Added virtual memory back (should help to launch N64 titles)

2023/06/02
* Updated ``koriki-rg35xx_20230602.img.gz``
* Fixed battery charge indicator

2023/06/01
* Reworked system settings menu
* Added system overclock support
* Added cores back to the image
* Added battery charger logic (use reboot or shutdown in simplemenu to change behavior)
* System should work with either 1 sdcard or 2 sdcard (SD1 koriki, SD2 roms)
  * By default roms use the batocera roms names and assume those are located in ``/userdata/roms/<system>``
  * You can use garlic or OFW images for the roms in the second SDCARD slot, but you will need to modify the paths in ``/userdata/system/.simplemenu/sections_groups``
* SDCard partition auto expand is current broken

2023/05/31
* ~~Added koriki_share_rg35xx.zip archive. This archive is intended to be used for a second sdcard. Uncompress at the top of the archive.~~
* It contains the cores, as well as the folder structure to add roms to the system

2023/05/30
* First development release
* Image automatically generated
* SDCARD expansion
* External SDCARD compatibility
* Only a few set of cores included

## Working:
* Navigation, rom selection, launching roms
* Brightness control
* Battery indicator
* Initial volume control via keys (menu does not reflect control change)
* Volume control for all emulators (thank to [adixal](https://codeberg.org/adixal) for this)
* System overclock control
* Retroarch menu can be opened with the ``Menu`` key. Quit with ``START``+``SELECT``
* Content auto-copied when a new SDCARD is inserted
* Auto mounts last partition of a second sdcard if present
* USB ethernet
* External controllers (in retroarch)

## Not working:
* ~~By default the system will start showing a message saying "No Games Found". Just press B to go back and select your rom. This is a bug that needs to be fixed in this version of simplemenu~~
* Some menus, settings are still incomplete
* Some emulators need to be updated
* ~~Standalone emulators require key mappings to be updated~~
* Input event issues with retroarch (map retroarch configuration manually)
* Application launchers (e.g. Wifi, need to be updated)
* ~~When the console starts or the menu is opened after a game is closed simplemenu shows there are no games available. Press B to go back and then you can go into the selection.~~
* ~~Controls for standalone emulators are not properly mapped~~
* ~~Networking is still not configured~~
* ~~External controllers are not working at this point (kernel is in adb mode by default)~~
* Selecting a core in simplemenu changes the background color (mostly cosmetic)

## Tweaking configuration:
* HOME is ``/userdata/system`` (also alias to /mnt/SDCARD)
* Simplemenu files are in ``$HOME/.simplemenu``
* Sections are in ``$HOME/.simplemenu/section_groups`` (self explanatory)
* Emulator launchers are in ``$HOME/.simplemenu/launchers`` (self explanatory, but look for the ones like neogeo_libretro that have been already adapted) 
* By default the firmware expects batocera-like roms partitions (with folders under roms like neogeo, mame, megadrive, etc.). It's possible add additional roms paths in ``$HOME/.simplemenu/section_groups`` .ini files. For instance, for master system you can see an example of batocera path (/userdata/roms/mastersystem) and Miyoo Koriki path (/userdata/Roms/MS). 
```[MASTER SYSTEM]
execs = /userdata/system/.simplemenu/launchers/picodrive_libretro,/userdata/system/.simplemenu/launchers/genesis_plus_gx_libretro,/userdata/system/.simplemenu/launchers/gearsystem_libretr
romDirs = /userdata/roms/mastersystem/,/userdata/Roms/MS
romExts = .zip,.sms,.7z
```
* The snapshots/art/thumbnail previews are by default under ``media/images`` (e.g. neogeo/media/images/mslug.png). You can change that path in $HOME/.simplemenu/config.ini and change the value of ``media_folder = media/images`` (for instance ``media_folder = Img`` is the default Koriki/miyoo path).

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v0.8'>Koriki for RG35XX Beta 0.8</a>.</em>

**Comment by cyberbol** [comment]
Hi
I love koriki. But i have some issue. I was used GarlicOS on two SD cards. All roms are on sdcard2. When I ijnstalled koroki on sd1 now some of the rom from sd2 are visible, nbut not all. I think its problems that koriki using different name of folders for platform, but I cant find anywhere what names should be. i mean i need compare them with garlicos folder names.

Also on garlicos default cover.png are in Imgs folder in each platform folders.  I dont know how to make them visible in koriki

**Reply by michaelgregorio** [reply]
media/images is the default in koriki

**Reply by tsu-kunn** [reply]
Place a png file with the same rom name in the following path and you will see the cover image.

roms/XXX/media/images

XXXX = GB, GBC, GBA, ...

If you share with GarlicOS, you will have double management.

**Comment by Zyglarz** [comment]
Hi

I've tested all custom OS so far but only GarlicOS worked fine for me . Dunno why performance on Koriki is kinda poor . PSX games like Tekken 3 havin drops and audio issues . Even with OC .  Tried both SD cards . Same problems . 

i've formated SD card , done fresh install using Balena Etcher like in tutorial . All goin smooth and quick . Do i need to do some fix or change settings before I start game ? Videos on YT showing better performance than mine RG35xx

**Reply by michaelgregorio** [reply]
Just turn off the vertical sync in retroarch setting
